cpp/tests/test_toppra_approach.cpp

```diff
@@ -0,0 +1,164 @@
+  if(printInfo) std::cout << "path_acc_\n " << path_acc_ << std::endl;
+  if(printInfo) std::cout << "times \n " << times.transpose() << std::endl;
+
+  //#### create linear joint-space constraints ####
+  Eigen::VectorXd velLimitLower = Eigen::VectorXd::Zero(numJoints);
+  Eigen::VectorXd velLimitUpper = Eigen::VectorXd::Zero(numJoints);
+  Eigen::VectorXd accLimitLower = Eigen::VectorXd::Zero(numJoints);
+  Eigen::VectorXd accLimitUpper = Eigen::VectorXd::Zero(numJoints);
+  velLimitLower << -1, -0.5;
+  velLimitUpper << 1, 0.5;
+  accLimitLower << -0.05, -0.1;
+  accLimitUpper << 0.2, 0.3;
+  std::shared_ptr<toppra::constraint::LinearJointVelocity> ljv;
+  ljv = std::make_shared<toppra::constraint::LinearJointVelocity>(velLimitLower,velLimitUpper);//[[-1, 1], [-0.5, 0.5]]
+  std::shared_ptr<toppra::constraint::LinearJointAcceleration> lja;
+  lja = std::make_shared<toppra::constraint::LinearJointAcceleration>(accLimitLower,accLimitUpper);//[[-0.05, 0.2], [-0.1, 0.3]]
+  std::vector<std::shared_ptr<toppra::LinearConstraint>> constraints;
+  constraints.push_back(ljv);
+  constraints.push_back(lja);
+
+  //#### create toppra ####
+  std::shared_ptr<toppra::algorithm::TOPPRA> algo;
+  algo = std::make_shared<toppra::algorithm::TOPPRA>(constraints, path);
+  const toppra::value_type vel_start = 0;
+  const toppra::value_type vel_end = 0;
+  toppra::ReturnCode rc1 = algo->computePathParametrization(vel_start, vel_end);
+  if(printInfo) std::cout << "rc1 = " << int(rc1) << std::endl;
+  ASSERT_EQ(rc1, toppra::ReturnCode::OK);
+  toppra::ReturnCode rc2 = algo->computeFeasibleSets();
+  if(printInfo) std::cout << "rc2 = " << int(rc2) << std::endl;
+  ASSERT_EQ(rc2, toppra::ReturnCode::ERR_FAIL_FEASIBLE); //TODO why not ok? => ASSERT_EQ(rc2, toppra::ReturnCode::OK);
+  toppra::ParametrizationData pd = algo->getParameterizationData();
+  if(printInfo) std::cout << "pd.gridpoints \n " << pd.gridpoints.transpose() << std::endl;
+  if(printInfo) std::cout << "pd.parametrization \n " << pd.parametrization.transpose() << std::endl;
+  if(printInfo) std::cout << "pd.controllable_sets \n " << pd.controllable_sets.transpose() << std::endl;
+  if(printInfo) std::cout << "pd.feasible_sets \n " << pd.feasible_sets.transpose() << std::endl;
+  if(printInfo) std::cout << "pd.ret_code = " << int(pd.ret_code) << std::endl;
+  ASSERT_EQ(pd.ret_code, toppra::ReturnCode::OK);
+
+  //#### create constant accelaration parametrizer ####
+  Eigen::VectorXd gridpoints = pd.gridpoints; //Grid-points used for solving the discretized problem.
+  Eigen::VectorXd vsquared = pd.parametrization; //Output parametrization (squared path velocity)
+  std::shared_ptr<toppra::parametrizer::ConstAccel> ca;
+  ca = std::make_shared<toppra::parametrizer::ConstAccel>(path, gridpoints, vsquared);
+  if(printInfo) std::cout << "ca->validate() = " << ca->validate() << std::endl;
+  ASSERT_FALSE(ca->validate()); //TODO why false? => ASSERT_TRUE(ca->validate());
```

Comment:
This should not be false. Look like there is a bug in the parametrizer class, or this instance is not feasible.

Reply:
I fully agree, but I don't understand how to fix it.